pypesto/sampling/geweke_test.py

```diff
@@ -0,0 +1,206 @@
+import numpy as np
+from scipy.stats import norm
+
+
+def spectrum(x: np.array,
+             nfft: int = None,
+             nw: int = None):
+    """
+    Power spectral density using Hanning window.
+
+    Parameters
+    ----------
+    x:
+        Fraction/fragment of one single parameter
+        of the MCMC chain.
+    nfft:
+        The n-point discrete Fourier Transform.
+    nw:
+        Number of windows.
+
+    Returns
+    -------
+    spectral_density:
+        The spectral density.
+
+    """
+
+    if nfft is None:
+        nfft = np.min(len(x), 256)
+
+    if nw is None:
+        nw = np.floor(nfft / 4).astype(int)
+
+    n_overlap = np.floor(nw / 2).astype(int)
+
+    # Hanning window
+    w = .5 * (1 - np.cos(
+        2 * np.pi * np.transpose(np.arange(1, nw+1)) /
+        (nw + 1)))
+    n = len(x)
+    if n < nw:
+        x[nw] = 0
+        n = nw
+
+    # Number of windows
+    k = np.floor((n - n_overlap) / (nw - n_overlap)).astype(int)
+    index = np.arange(nw)
+    # Normalizing scale factor
+    kmu = k * np.linalg.norm(w) ** 2
+    spectral_density = np.zeros((nfft))
+
+    for i in range(k):
+        xw = w * x[index]
+        index += (nw - n_overlap)
+        Xx = np.absolute(np.fft.fft(xw, n=nfft, axis=0)) ** 2
+        spectral_density += Xx
+
+    # Normalize
+    spectral_density = spectral_density * (1 / kmu)
+
+    n2 = np.floor(nfft / 2).astype(int)
+    spectral_density = spectral_density[0:n2]
+
+    return spectral_density
+
+
+def spectrum0(x: np.array):
+    """
+    Calculates the spectral density at frequency zero.
+
+    Parameters
+    ----------
+    x:
+        Fraction/fragment of the MCMC chain.
+
+    Returns
+    -------
+    spectral_density_zero:
+        Spectral density at zero.
+
+    """
+
+    n_samples, n_par = x.shape
+    spectral_density_zero = np.zeros((1, n_par))
+
+    for i in range(n_par):
+        spectral_density_zero[:, i] = \
+            spectrum(x[:, i], n_samples)[0]
+    return spectral_density_zero
+
+
+def calculate_zscore(chain: np.array,
+                     a: float = 0.1,
+                     b: float = 0.5):
```

Comment:
```suggestion
                     b: float = 0.5) -> Tuple[float, float]:
```
where `from typing import Tuple`